File: src/main/java/com/gmail/borlandlp/minigamesdtools/activepoints/reaction/custom/TeamIncrementWinTicketsReactionCreator.java
package com.gmail.borlandlp.minigamesdtools.activepoints.reaction.custom;

import com.gmail.borlandlp.minigamesdtools.MinigamesDTools;
import com.gmail.borlandlp.minigamesdtools.activepoints.ActivePoint;
import com.gmail.borlandlp.minigamesdtools.activepoints.reaction.Reaction;
import com.gmail.borlandlp.minigamesdtools.config.ConfigPath;
import com.gmail.borlandlp.minigamesdtools.creator.AbstractDataProvider;
import com.gmail.borlandlp.minigamesdtools.creator.Creator;
import com.gmail.borlandlp.minigamesdtools.creator.CreatorInfo;
import org.bukkit.configuration.ConfigurationSection;

import java.util.ArrayList;
import java.util.List;

@CreatorInfo(creatorId = "team_increment_win_ticket_reaction")
public class TeamIncrementWinTicketsReactionCreator extends Creator {
    @Override
    public Reaction create(String id, AbstractDataProvider dataProvider) throws Exception {
        TeamIncrementWinTicketsReaction reaction = new TeamIncrementWinTicketsReaction();
        ConfigurationSection configurationSection = MinigamesDTools.Companion.getInstance().getConfigProvider().getEntity(ConfigPath.ACTIVE_POINT_REACTIONS, id).getData();
        reaction.setActivePoint((ActivePoint) dataProvider.get("active_point_instance"));
        reaction.setValue(Integer.parseInt(configurationSection.get("value").toString()));

        return reaction;
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/activepoints/reaction/custom/TeamIncrementWinTicketsReactionCreator.kt
package com.gmail.borlandlp.minigamesdtools.activepoints.reaction.custom

import com.gmail.borlandlp.minigamesdtools.MinigamesDTools.Companion.instance
import com.gmail.borlandlp.minigamesdtools.activepoints.ActivePoint
import com.gmail.borlandlp.minigamesdtools.activepoints.reaction.Reaction
import com.gmail.borlandlp.minigamesdtools.config.ConfigPath
import com.gmail.borlandlp.minigamesdtools.creator.AbstractDataProvider
import com.gmail.borlandlp.minigamesdtools.creator.Creator
import com.gmail.borlandlp.minigamesdtools.creator.CreatorInfo

@CreatorInfo(creatorId = "team_increment_win_ticket_reaction")
class TeamIncrementWinTicketsReactionCreator : Creator() {
    override val dataProviderRequiredFields: List<String>
        get() = listOf("active_point_instance")

    @Throws(Exception::class)
    override fun create(id: String, dataProvider: AbstractDataProvider): Reaction {
        return TeamIncrementWinTicketsReaction().apply {
            this.activePoint = dataProvider["active_point_instance"] as ActivePoint
            val configEntity = instance!!.configProvider!!.getEntity("active_point_reactions", id)!!
            this.value = configEntity.data["value"].toString().toInt()
        }
    }
}